lib/gpprmsya.gi

```diff
@@ -808,45 +817,214 @@ function ( G )
-    sgn := 1;
-    for i  in [1..deg-1] do
-        k := Random( [ i .. deg] );
-        tmp := rnd[i];
-        rnd[i] := rnd[k];
-        rnd[k] := tmp;
-    od;
+    rnd := FLOYDS_ALGORITHM(deg, true);
 
     # return the permutation
     return PermList( rnd )^MappingPermListList([1..deg],mov);
 end);
 
+
 #############################################################################
 ##
-#M  StabilizerOp( <nat-sym-grp>, <int>, OnPoints )
+#M  Size( <nat-alt-grp> )
 ##
-InstallOtherMethod( StabilizerOp,"symmetric group", true,
-    [ IsNaturalSymmetricGroup, IsPosInt, IsFunction ],
-  # the objects might be a group element: rank up	
-  RankFilter(IsMultiplicativeElementWithInverse),
-
-function( sym, p, opr )
-    # test for internal rep
-    if HasGeneratorsOfGroup(sym) and 
-      not ForAll(GeneratorsOfGroup(sym),IsInternalRep) then
-      TryNextMethod();
+InstallMethod( Size,
+    true,
+        [ IsNaturalAlternatingGroup ], 0,
+        function(a)
+    local n;
+    n := NrMovedPoints(a);
+    if n <= 2 then
+        return 1;
     fi;
+    return Factorial( n )/2;
+end);
+
 
-    if opr <> OnPoints  then
+#############################################################################
+##
+#M Intersection2
+
+InstallMethod( Intersection2, [IsNaturalSymmetricGroup, IsNaturalSymmetricGroup],
+        function(s1,s2)
+    return SymmetricGroup(Intersection(MovedPoints(s1),MovedPoints(s2)));
+end);
+
+InstallMethod( Intersection2, [IsNaturalSymmetricGroup, IsNaturalAlternatingGroup],
+        function(s1,s2)
+    return AlternatingGroup(Intersection(MovedPoints(s1),MovedPoints(s2)));
+end);
+
+InstallMethod( Intersection2, [IsNaturalAlternatingGroup, IsNaturalSymmetricGroup],
+        function(s1,s2)
+    return AlternatingGroup(Intersection(MovedPoints(s1),MovedPoints(s2)));
+end);
+
+InstallMethod( Intersection2, [IsNaturalAlternatingGroup, IsNaturalAlternatingGroup],
+        function(s1,s2)
+    return AlternatingGroup(Intersection(MovedPoints(s1),MovedPoints(s2)));
+end);
+
+
+InstallMethod( Intersection2, [IsNaturalSymmetricGroup, IsPermGroup],
+        function(s,g)
+    return Stabilizer(g,Difference(MovedPoints(g),MovedPoints(s)), OnTuples);
+end);
+
+InstallMethod( Intersection2, [IsPermGroup, IsNaturalSymmetricGroup],
+        function(g,s)
+    return Stabilizer(g,Difference(MovedPoints(g),MovedPoints(s)), OnTuples);
+end);
+
+InstallMethod( Intersection2, [IsNaturalAlternatingGroup, IsPermGroup],
+        function(s,g)
+    return AlternatingSubgroup(Stabilizer(g,Difference(MovedPoints(g),MovedPoints(s)), OnTuples));
+end);
+
+InstallMethod( Intersection2, [IsPermGroup, IsNaturalAlternatingGroup],
+        function(g,s)
+    return AlternatingSubgroup(Stabilizer(g,Difference(MovedPoints(g),MovedPoints(s)), OnTuples));
+end);
+
+
+        
+    
+    
+
+
+
+
+#############################################################################
+##
+#M  StabilizerOp( <nat-sym-grp>, ...... )
+##
+SYMGP_STABILIZER := function(sym, arg)
```

Comment:
@stevelinton while loading the library, `SYMGP_STABILIZER` causes "unbound global variable" warning for `gens` and `size`. This does not crash testinstall and teststandard, but hits the test of Tutorial examples. Shall I just add them to `local` ?